x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureaistudio/AzureAiStudioServiceSettings.java

```diff
@@ -51,63 +52,60 @@ protected static BaseAzureAiStudioCommonFields fromMap(
             AzureAiStudioService.NAME,
             context
         );
-        AzureAiStudioEndpointType endpointType = extractRequiredEnum(
+        AzureAiStudioDeploymentType deploymentType = extractRequiredEnum(
             map,
-            ENDPOINT_TYPE_FIELD,
+            DEPLOYMENT_TYPE_FIELD,
             ModelConfigurations.SERVICE_SETTINGS,
-            AzureAiStudioEndpointType::fromString,
-            EnumSet.allOf(AzureAiStudioEndpointType.class),
+            AzureAiStudioDeploymentType::fromString,
+            EnumSet.allOf(AzureAiStudioDeploymentType.class),
             validationException
         );
-
-        AzureAiStudioProvider provider = extractRequiredEnum(
+        String model = extractOptionalString(
             map,
-            PROVIDER_FIELD,
+            MODEL_FIELD,
             ModelConfigurations.SERVICE_SETTINGS,
-            AzureAiStudioProvider::fromString,
-            EnumSet.allOf(AzureAiStudioProvider.class),
             validationException
         );
 
-        return new BaseAzureAiStudioCommonFields(target, provider, endpointType, rateLimitSettings);
+        return new BaseAzureAiStudioCommonFields(target, deploymentType, model, rateLimitSettings);
     }
 
     protected AzureAiStudioServiceSettings(StreamInput in) throws IOException {
         this.target = in.readString();
-        this.provider = in.readEnum(AzureAiStudioProvider.class);
-        this.endpointType = in.readEnum(AzureAiStudioEndpointType.class);
+        this.deploymentType = in.readEnum(AzureAiStudioDeploymentType.class);
+        this.model = in.readString();
```

Comment:
Similar rationale as down below. I think this needs to be `in.readOptionalString()`